### PR TITLE
Increase timeout on Sql serverConnection

### PR DESCRIPTION
**Pull request by rbans96**
## Description
This PR sets the explicit statement timeout on ServerConnection. The default timeout is 10 mins which is not enough for few long running statements in the migration scripts.

## Related issues
Addresses [#86103](https://microsofthealth.visualstudio.com/Health/_boards/board/t/Olympus/Stories/?workitem=86103)

## Testing
This is tested manually by upgrading the schema for 1TB database from schema version 17 to 19.

## Semver Change ([docs](https://github.com/microsoft/healthcare-shared-components/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking

### CODE REVIEW

File: src/Microsoft.Health.SqlServer/Configs/SqlServerDataStoreConfiguration.cs [hunk 1]
```diff
@@ -51,5 +51,10 @@ public class SqlServerDataStoreConfiguration
         /// If set, the client id of the managed identity to use when connecting to SQL, if AuthenticationType == ManagedIdentity.
         /// </summary>
         public string ManagedIdentityClientId { get; set; }
+
+        /// <summary>
+        /// Specifies the statement timeout to set on ServerConnection.
+        /// </summary>
+        public int StatementTimeoutInSecs { get; set; } = 14400;
```
Comment (brendankowitz) [inline src/Microsoft.Health.SqlServer/Configs/SqlServerDataStoreConfiguration.cs:58]: I think a TimeSpan would be better, but check if that works with config as expected

File: src/Microsoft.Health.SqlServer/Features/Schema/Manager/SchemaManagerDataStore.cs [hunk 2]
```diff
@@ -45,7 +54,13 @@ public async Task ExecuteScriptAndCompleteSchemaVersionAsync(string script, int
                 }
 
                 var server = new Server(serverConnection);
+                var watch = Stopwatch.StartNew();
+                _logger.LogInformation($"Script execution started at {DateTime.UtcNow}");
```
Comment (brendankowitz) [inline src/Microsoft.Health.SqlServer/Features/Schema/Manager/SchemaManagerDataStore.cs:58]: nit: `Clock.UtcNow`

File: src/Microsoft.Health.SqlServer/Features/Schema/Manager/SchemaManagerDataStore.cs [hunk 3]
```diff
@@ -146,5 +161,13 @@ public async Task<bool> InstanceSchemaRecordExistsAsync(CancellationToken cancel
             using var command = new SqlCommand(procedureQuery, connection);
             return (int)await command.ExecuteScalarAsync(cancellationToken).ConfigureAwait(false) != 0;
         }
+
+        private ServerConnection GetServerConnectionWithTimeout(SqlConnection sqlConnection)
+        {
+            var serverConnection = new ServerConnection(sqlConnection);
+            serverConnection.StatementTimeout = _sqlServerDataStoreConfiguration.StatementTimeoutInSecs;
```
Comment (bbatjargal) [inline src/Microsoft.Health.SqlServer/Features/Schema/Manager/SchemaManagerDataStore.cs:168]: Just for my understanding, Is serverConnection.StatementTimeout same as server.ConnectionContext.StatementTimeout (updating it before line 60)? For my test, I used "server.ConnectionContext.StatementTimeout".

File: src/Microsoft.Health.SqlServer/Configs/SqlServerDataStoreConfiguration.cs (hunk 1 above)
Comment (bbatjargal) [inline src/Microsoft.Health.SqlServer/Configs/SqlServerDataStoreConfiguration.cs:58]: Is this config accessible from the schema manager command-line tool and the schema manager Paas?

File: src/Microsoft.Health.SqlServer/Features/Schema/Manager/SchemaManagerDataStore.cs (hunk 3 above)
Comment (rbans96) [reply src/Microsoft.Health.SqlServer/Features/Schema/Manager/SchemaManagerDataStore.cs]: Yes, statementTimeout is property on serverConnection only. When we set StatementTimeout to server.ConnectionContext, its also adding timeout to serverConnection itself.

File: src/Microsoft.Health.SqlServer/Configs/SqlServerDataStoreConfiguration.cs (hunk 1 above)
Comment (rbans96) [reply src/Microsoft.Health.SqlServer/Configs/SqlServerDataStoreConfiguration.cs]: This config is for oss command line tool. Once these changes gets merged and in a version upgrade PR for healthcare-shared-components nugets in health-paas, we will need to add this parameter [here](https://microsofthealth.visualstudio.com/Health/_git/health-paas?path=/src/SqlManagementApplication/SchemaManager.Paas/Commands/ApplyPaasCommand.cs&version=GBmaster&line=86&lineEnd=87&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents)

Comment (rbans96) [reply src/Microsoft.Health.SqlServer/Configs/SqlServerDataStoreConfiguration.cs]: Thanks, updated!

**Review by bbatjargal — APPROVED** [review]

**Pull request merged**